src/views/SearchWikipedia.vue

```diff
@@ -1,37 +1,37 @@
 <template>
     <div class="view search-wp">
-        <form @submit="onSubmit($event, query)">
+        <form @submit="onSubmit">
+            <router-link to='Story'><div class="back"/></router-link>
             <div class="label">{{ $i18n('search-box') }}</div>
-            <input class="query" type="text" :placeholder="[[ $i18n('search-box') ]]" v-model="query"/>
+            <input class="query" type="text" :placeholder="[[ $i18n('search-box') ]]" v-model="query" required/>
             <input class="submit" type="submit" :value="[[ $i18n('search-text') ]]"/>
+            <div class="icon"/>
+            <div v-if="query" class="close"  v-on:click="onClear"/>
         </form>
-        <div class="results">
-            <div class="result" v-for="result in searchResults" :key="result.title">
-                <router-link :to="{ name: 'BrowseArticle', params: { article: result.title } }">
-                  <div class="img" :style="{backgroundImage: `url(${result.thumb})` }"/>
-                  <div class="info">
-                    <div class="title">{{result.title}}</div>
-                    <div class="description">{{result.desc}}</div>
-                  </div>
-                </router-link>
-            </div>
-        </div>
+        <ListView :items="searchResults"/>
     </div>
 </template>
 <script>
     import { mapActions, mapGetters } from 'vuex'
+    import ListView from '@components/ListView.vue'
     export default {
       name: 'SearchWikipedia',
+      components: { ListView },
       data: () => {
         return {
           query: ''
         }
       },
       methods: {
         ...mapActions(['search']),
-        onSubmit: function(e, query) {
+        onSubmit: function(e) {
+          const queryString = this.query.trim()
           e.preventDefault()
-          this.search(query)
+          this.search(queryString)
+        },
+        onClear: function(e) {
+          e.preventDefault();
+          this.query = '';
```

Comment:
do you mean 86b0381345c38674bafd2a1333695b41b5caa7ab? the query is the internal state in Search, therefore staying there.

Reply:
Yes.

It could be interesting to bring the query in the store so we can recreate the state correctly if we navigate back to the search page but now a blocker right now.

Reply:
Actually, let's do it now. It's weird when you go back to the search page. You have results but no query and no x button to clear it.

Reply:
Yup, make sense, we want to keep the query when back to search page, update here 30dd0d8c4fb8b915b0dd31e029a2f74c5e200080